imu_integrator/plugin.cpp

```diff
@@ -0,0 +1,479 @@
+#include <chrono>
+#include <iomanip>
+#include <thread>
+#include <eigen3/Eigen/Dense>
+
+#include "../open_vins/ov_msckf/src/state/Propagator.h"
+#include "../open_vins/ov_core/src/types/IMU.h"
+
+#include "common/plugin.hpp"
+#include "common/switchboard.hpp"
+#include "common/data_format.hpp"
+#include "common/threadloop.hpp"
+
+using namespace ILLIXR;
+
+class imu_integrator : public threadloop {
+public:
+	imu_integrator(std::string name_, phonebook* pb_)
+		: threadloop{name_, pb_}
+		, sb{pb->lookup_impl<switchboard>()}
+		, _m_imu_cam{sb->subscribe_latest<imu_cam_type>("imu_cam")}
+		, _m_in{sb->subscribe_latest<imu_integrator_seq>("imu_integrator_seq")}
+		, _m_imu_integrator_input{sb->subscribe_latest<imu_integrator_input>("imu_integrator_input")}
+		, _m_imu_raw{sb->publish<imu_raw_type>("imu_raw")}
+		, _seq_expect(1)
+	{}
+
+	virtual skip_option _p_should_skip() override {
+		auto in = _m_in->get_latest_ro();
+		if (!in || in->seq == _seq_expect-1) {
+			// No new data, sleep to keep CPU utilization low
+			std::this_thread::sleep_for(std::chrono::milliseconds{1});
```

Comment:
What exactly does this sleep? Which thread?  We probably shouldn't be sleeping the core runtime loop for an entire 1ms...? If this is just sleeping this components' loop, then that's fine.